Assets/Scripts/Framework/Train/TrainStationController.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        private void Update()
+        {
+            if (!IsServer || TrainEngine.Instance.EngineState == TrainEngineState.Inactive)
+                return;
+
+            _canPark = ParkCheck(out int nearestStationIndex);
+
+            if(!_canPark && _isParked)
+            {
+                UnparkTrain();
+                return;
+            }
+
+            if (Mathf.Abs(TrainEngine.Instance.CurrentSpeed) <= 0.005f && !_isParked)
+            {
+                if (TrainEngine.Instance.CurrentGearIndex != 0 || !_canPark)
+                    return;
+
+                ParkTrainAtClosestStation(nearestStationIndex);
+            }
+
+            else if (Mathf.Abs(TrainEngine.Instance.CurrentSpeed) >= _minTrainSpeedToPark && _isParked)
+                UnparkTrain();
+        }
+
+        [Server]
+        private bool ParkCheck(out int nearestStationIndex)
+        {
+            Vector3 trainPosition = _trainController.Spline.EvaluatePosition(_trainController.DistanceAlongSpline);
+            nearestStationIndex = StationManager.Instance.GetNearestStationIndex(trainPosition, out _);
+
+            StationCameraBlendingContainer closestStation = StationManager.Instance.StationContainers[nearestStationIndex];
+
+            bool min = closestStation.StationBoundingBoxCollider.bounds.Contains(_minimumPoint.position);
+            bool max = closestStation.StationBoundingBoxCollider.bounds.Contains(_maximumPoint.position);
+
+            return min && max;
+        }
+
+        [ServerRpc(RequireOwnership = false)]
+        private void UnparkTrain() => UnparkTrainObserver();
+
+
+        [ServerRpc(RequireOwnership = false)]
+        private void ParkTrainAtClosestStation(int nearestStationIndex) => ParkTrain(nearestStationIndex);
+
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
+        private void ParkTrain(int closestStationIndex)
+        {
+            SetIsParked(true);
+
+            CinemachineVirtualCamera nearestStationCamera 
+                = StationManager.Instance.StationContainers[closestStationIndex].StationCamera;
+            
+            CameraManager.Instance.ChangeActiveCamera(nearestStationCamera);
+            
+            _currentStationAnimator = nearestStationCamera.transform.parent.GetComponent<Animator>();
+            _currentStationAnimator.SetTrigger(_enterAnimationHash);
+        }
+
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
+        private void UnparkTrainObserver()
+        {
+            SetIsParked(false);
+
+            if (_currentStationAnimator != null )
+                _currentStationAnimator.SetTrigger(_exitAnimationHash);
+
+            CameraManager.Instance.ChangeActiveCamera(CameraManager.Instance.TrainCamera);
+            _currentStationAnimator = null;
+        }
+
+        private void SetIsParked(bool state)
+        {
+            _isParked = state;
+            OnParkStateChanged?.Invoke(state);
+        }
```

Comment:
Instead of calling the event in a method you can make a public getter and private setter for `IsParked` and call the event inside the setter.